theme/templates/resource-landing-page/title-section.html

```diff
@@ -51,6 +51,12 @@
                         <br>
                         <span>Content files are required to publish the resource or make the resource public or discoverable.</span>
                 {% endif %}
+                {%  if cm.resource_type|lower == "collectionresource" %}
+                    {% if not cm.has_resources%}
+                        <br>
+                        <span>Collection should contain at least one resource to be made it published, public or discoverable.</span>
+                    {% endif %}
+                {% endif %}
```

Comment:
I think the logic is not correct here. For example, when testing on playground, after filling in abstract and keyword subject, the message shows up indicating it is sufficient to make the resource public or publish it. I think you should override has_required_content_files() function to make sure collection should contain at least one resource (or I think at least two resources is better). Then you don't have to add specific logic for collection resource, but rather the generic can_be_public function will be sufficient since it checks metadata and content files.